src/domains/point_set.jl

```diff
@@ -25,34 +27,44 @@ julia> PointSet([(rand(),rand()) for i in 1:100])
 ```
 """
 struct PointSet{T,N} <: SpatialDomain{T,N}
-  coords::Matrix{T}
+  coords::Vector{SVector{N,T}} 
 
   function PointSet{T,N}(coords) where {N,T}
     @assert !isempty(coords) "coordinates must be non-empty"
     new(coords)
   end
 end
 
-PointSet(coords::AbstractMatrix{T}) where {T} =
-  PointSet{T,size(coords,1)}(coords)
+function PointSet(coords::AbstractMatrix{T}) where {T}
+  N = size(coords, 1)
+  points = Vector{SVector{N,T}}()
+  for row in eachcol(coords)
+    push!(points,SVector{N,T}(row))
+  end
+  PointSet{T,N}(points)
+end
 
-PointSet(coords::AbstractVector{NTuple{N,T}}) where {N,T} =
-  PointSet([c[i] for i in 1:N, c in coords])
+function PointSet(coords::AbstractVector{NTuple{N,T}}) where {N,T}
+  points = Vector{SVector{N,T}}()
+  for row in coords
+    push!(points,SVector{N,T}(row))
+  end
+  PointSet{T,N}(points)
```

Comment:
After you remove the inner constructor, perhaps this outer constructor won't be needed. In case it is needed, please use the following:

```julia
PointSet(coords::AbstractVector{<:NTuple}) = PointSet(SVector.(coords))
```

Reply:
Much better indeed

Reply:
I need to inject the size of the point (N):

PointSet(coords::AbstractVector{<:NTuple}) = PointSet(SVector{N}.(coords))

Do you know a trick to do so without writing a function with two lines?

Reply:
It seems if we have:
```
function PointSet(coords::AbstractVector{<:AbstractVector})
  @assert !isempty(coords) "coordinates must be non-empty"
  @assert length(coords) > 0 "coordinates must contain one point at least"
  N = length(coords[1]) #getting 
  PointSet(SVector{N}.(coords))
end
```
Then
```
PointSet(coords::AbstractMatrix) = PointSet(collect(eachcol(coords)))
PointSet(coords::AbstractVector{<:NTuple}) = PointSet(collect(map(x -> Vector(collect(x)),coords)))
```
works well

Reply:
I don't understand why the original version `PointSet(SVector.(coords))` didn't do the job. Why we need all these lines of code in the constructor above?

Reply:
Simply because it does not compile!
SVector needs somehow the size of the static array (N). SVector{N}.(coords)

Reply:
Strange, it works for me:

```julia
julia> v = [(1.,2.),(3.,4.)]
2-element Array{Tuple{Float64,Float64},1}:
 (1.0, 2.0)
 (3.0, 4.0)
julia> SVector.(v)
2-element Array{SArray{Tuple{2},Float64,1,2},1}:
 [1.0, 2.0]
 [3.0, 4.0]
```
Let me check with the full signature...

Reply:
Perhaps this is about the Julia version? I am on Julia v1.5

Reply:
I'm using v1.5.1

Reply:
It works for that case, but not in this case:
```
a = [[1.0, 2.0],[10.,20.]]
SVector.(a)
```

Reply:
Yes, we don't want to support `AbstractVector{<:AbstractVector}`. Tuples and static vectors have known length at compile time, unlike standard vectors.

At the end we should have 3 constructors:
- Inner constructor for list of SVector which is created by the compiler
- Outer constructor for list of Tuple which calls the inner constructor after a `SVector.(v)` conversion
- Outer constructor for 2D matrix which uses the number of rows as `N` in `SVector{N}`

Reply:
I think that is lack of flexibility for uses. Using vector of vector is quite common.
Interestingly, if you use a matrix and iterate the columns, you get a Vector (View) for each column not a tuple.
```

julia> a = rand(2,4)
2×4 Array{Float64,2}:
 0.409933  0.427235  0.487495  0.327178
 0.316636  0.845672  0.424418  0.581679

julia> first(eachcol(a))
2-element view(::Array{Float64,2}, :, 1) with eltype Float64:
 0.40993283206458475
 0.3166363584400025
```

Reply:
> Yes, we don't want to support `AbstractVector{<:AbstractVector}`. Tuples and static vectors have known length at compile time, unlike standard vectors.
> 
> At the end we should have 3 constructors:
> 
>     * Inner constructor for list of SVector which is created by the compiler
> 
>     * Outer constructor for list of Tuple which calls the inner constructor after a `SVector.(v)` conversion
> 
>     * Outer constructor for 2D matrix which uses the number of rows as `N` in `SVector{N}`

Vector of vector and Vector of tuple are quite the same from user perspective. The important thing is we will normilised them to an internal representation that is much more performant. Also, both representations may fail if the length of each point are different. The current implementation assumes that the final point length will be equal to the first point length. Matrix does not have that issue.